pkg/cmd/pr/merge/merge.go

```
@@ -192,6 +190,16 @@ func (m *mergeContext) disableAutoMerge() error {
 	return m.infof("%s Auto-merge disabled for pull request #%d\n", m.cs.SuccessIconWithColor(m.cs.Green), m.pr.Number)
 }
 
+// Check if this pull request is in a merge queue
+func (m *mergeContext) inMergeQueue() error {
+	// if the pull request is in a merge queue no further action is possible
+	if m.pr.IsInMergeQueue {
+		_ = m.warnf("%s Pull request #%d is already queued to merge\n", m.cs.WarningIcon(), m.pr.Number)
+		return cmdutil.SilentError
```

SilentError is going to cause the process to exit with a nonzero status. Sometimes, we choose to exit the process with zero status even if a warning was printed. We have no clear-cut rules for this, but we usually make a judgment along the lines of: if the end state that the user has wanted was accomplished, gh should probably exit with zero status.

Since `gh pr merge 123` could be considered "successful" if `#123` is already queued to merge, I'm leaning towards having the process exit with success status. But we could also keep it like this. Thoughts?

---

I agree with that. The end result here is that the PR is in the queue, which may not have been what the user intended (they may not even know they are required to use a merge queue) but it is a successful end state. It won't merge immediately but should merge eventually in that scenario.

---

Okay! Then you can return an internal error type here, for example `errorAlreadyInMergeQueue`, and then on the command level you can ignore this error and just return `nil`.

---

done!

---

resolved in https://github.com/cli/cli/pull/5596/commits/f026a0aa3a8c5419507b9e13aa1a10317ab9c992